chain/pool/src/lib.rs

```diff
@@ -1,83 +1,120 @@
-use std::collections::btree_map::BTreeMap;
-use std::collections::HashMap;
+use std::collections::{HashMap, HashSet};
 
-use near_chain::ValidTransaction;
+use near_crypto::PublicKey;
 use near_primitives::transaction::SignedTransaction;
-use near_primitives::types::{AccountId, Nonce};
+use near_primitives::types::AccountId;
 
 pub use crate::types::Error;
+use near_primitives::hash::CryptoHash;
 
 pub mod types;
 
 /// Transaction pool: keeps track of transactions that were not yet accepted into the block chain.
 #[derive(Default)]
 pub struct TransactionPool {
-    num_transactions: usize,
-    /// Transactions grouped by account and ordered by nonce.
-    pub transactions: HashMap<AccountId, BTreeMap<Nonce, SignedTransaction>>,
+    /// Transactions grouped by a pair of (account ID, signer public key).
+    /// It's more efficient to keep transactions unsorted and with potentially conflicting nonce
+    /// than create a BTreeMap for every transaction on average.
```

Comment:
The comment seems outdated. It is not BTreeMap anymore?

Reply:
We've previously used BTreeMap instead of Vec.